dynet/lstm.cc

```diff
@@ -80,8 +82,44 @@ void LSTMBuilder::new_graph_impl(ComputationGraph& cg) {
 // layout: 0..layers = c
 //         layers+1..2*layers = h
 void LSTMBuilder::start_new_sequence_impl(const vector<Expression>& hinit) {
+  if (dropout_rate) { gal_dropout = true; } else { gal_dropout = false; }
```

Comment:
I'm not 100% clearn on this but I think this interacts counterintuitively with the inverted dropout semantics.

My experimental observation is that if I do `rnn.set_dropout(0.1)` my training loss doesn't go down at all (I assume because this means 10% retention rate) but if I do `rnn.set_dropout(0.0)` it converges (dropout is completely off), I think because this `if` doesn't trigger.

I imagine this won't be an issue when this is merged, if the old behaviour is not kept, but I figured this comment might be useful to others trying to use the code in this PR.